python/cudf/cudf/core/column/column.py

```diff
@@ -123,11 +123,26 @@ def mask_array_view(self):
     def __len__(self):
         return self.size
 
-    def to_pandas(self, index=None, **kwargs):
+    def to_pandas(self, index=None, nullable=False, **kwargs):
         if str(self.dtype) in NUMERIC_TYPES and self.null_count == 0:
-            pd_series = pd.Series(cupy.asnumpy(self.values))
+            pd_series = pd.Series(
+                cupy.asnumpy(self.values),
+                dtype=cudf.utils.dtypes.cudf_dtypes_to_pandas_dtypes.get(
+                    self.dtype, self.dtype
+                )
+                if nullable
+                else None,
+            )
         else:
             pd_series = self.to_arrow().to_pandas(**kwargs)
+            if (
+                nullable
+                and self.dtype
+                in cudf.utils.dtypes.cudf_dtypes_to_pandas_dtypes
+            ):
+                pd_series = pd_series.astype(
+                    cudf.utils.dtypes.cudf_dtypes_to_pandas_dtypes[self.dtype]
+                )
```

Comment:
I'm -1 on this approach. If someone passes `nullable=True` we should just always return the nullable type if appropriate regardless of the null count. Additionally, we shouldn't go cudf[int64] --> arrow[int64] --> pandas[float64] --> pandas[Int64]. We should either do cudf[int64] --> arrow[int64] --> pandas[Int64] or cudf[int64] --> device buffers --> host buffers --> pandas[Int64].

Reply:
You may consider using `pd.Int64Dtype.__from_arrow_()` and the equivalent for the other pandas dtypes:

```
In [10]: arrow_array = cudf.Series([1, None, 3]).to_arrow()

In [11]: pd.Int64Dtype().__from_arrow__(arrow_array)
Out[11]: 
<IntegerArray>
[1, <NA>, 3]
Length: 3, dtype: Int64
```
This pushes the responsibility for the conversion onto pandas at least.

Reply:
Changed the approach to use `ExtensionDtype.__from_arrow__`

Reply:
We could use `type_mapper` but it doesn't seem to work currently for pyarrow arrays: https://issues.apache.org/jira/browse/ARROW-9787

Reply:
Right - It works for `Table` but not `Array` IIRC.

It might be worth scoping out what it would take for us to add this to arrow ourselves, since nullable is the direction we're gravitating towards anyways.